examples/cim/gnn/actor.py

```diff
@@ -0,0 +1,367 @@
+from maro.simulator import Env
+from maro.simulator.scenarios.cim.common import Action, DecisionEvent
+from maro.utils import Logger, LogFormat, convert_dottable
+
+
+def organize_exp_list(experience_collections:dict, idx_mapping:dict):
+    """The function assemble the experience from multiple processes into a dictionary.
+
+    Args:
+         experience_collections (dict): It stores the experience in all agents. The structure is the same as what is
+            defined in the SharedStructure in the ParallelActor except additional key for experience length. For
+            example:
+
+            {
+                "len": numpy.array,
+                "s": {
+                    "v": numpy.array,
+                    "p": numpy.array,
+                }
+                "a": numpy.array,
+                "R": numpy.array,
+                "s_": {
+                    "v": numpy.array,
+                    "p": numpy.array,
+                }
+            }
+
+            Note that the experience from different agents are stored in the same batch in a sequential way. For
+            example, if agent x starts at b_x in batch index and the experience is l_x length long, the range [b_x,
+            l_x) in the batch is the experience of agent x.
+
+         idx_mapping (dict): The key is the name of each agent and the value is the starting index, e.g., b_x, of the
+            storage space where the experience of the agent is stored.
+    """
+    result = {}
+    tmpi = 0
+    for code, idx in idx_mapping.items():
+        exp_len = experience_collections["len"][0][tmpi]
+
+        s = organize_obs(experience_collections["s"], idx, exp_len)
+        s_ = organize_obs(experience_collections["s_"], idx, exp_len)
+        R = experience_collections["R"][idx:idx+exp_len]
+        R = R.reshape(-1, *R.shape[2:])
+        a = experience_collections["a"][idx:idx+exp_len]
+        a = a.reshape(-1, *a.shape[2:])
+
+        result[code] = {
+            "R": R,
+            "a": a,
+            "s": s,
+            "s_": s_,
+            "len": a.shape[0]
+        }
+        tmpi += 1
+    return result
+
+def organize_obs(obs, idx, exp_len):
+    """Helper function to transform the observation from multiple processes to a unified dictionary. """
+    tick_buffer, _, para_cnt, v_cnt, v_dim = obs["v"].shape
+    _, _, _, p_cnt, p_dim = obs["p"].shape
+    batch = exp_len * para_cnt
+    # v: tick_buffer, seq_len,  parallel_cnt, v_cnt, v_dim --> (tick_buffer, cnt, v_cnt, v_dim)
+    v = obs["v"][:, idx: idx+exp_len]
+    v = v.reshape(tick_buffer, batch, v_cnt, v_dim)
+    p = obs["p"][:, idx: idx+exp_len]
+    p = p.reshape(tick_buffer, batch, p_cnt, p_dim)
+    # vo: seq_len * parallel_cnt * v_cnt * p_cnt* --> cnt * v_cnt * p_cnt*
+    vo = obs["vo"][idx: idx+exp_len]
+    vo = vo.reshape(batch, v_cnt, vo.shape[-1])
+    po = obs["po"][idx: idx+exp_len]
+    po = po.reshape(batch, p_cnt, po.shape[-1])
+    vedge = obs["vedge"][idx: idx+exp_len]
```

Comment:
Be carefully please.

Reply:
Try fixing your flake8 may be helpful.

@ArthurJiang  Wenlei met some problem with the the command ` flake8 --config .github/linters/tox.ini`.